src/hotspot/share/gc/parallel/psScavenge.cpp

```diff
@@ -180,54 +180,68 @@ class PSKeepAliveClosure: public OopClosure {
 class PSEvacuateFollowersClosure: public VoidClosure {
  private:
   PSPromotionManager* _promotion_manager;
+  TaskTerminator* _terminator;
+  uint _worker_id;
+
  public:
-  PSEvacuateFollowersClosure(PSPromotionManager* pm) : _promotion_manager(pm) {}
+  PSEvacuateFollowersClosure(PSPromotionManager* pm, TaskTerminator* terminator, uint worker_id)
+    : _promotion_manager(pm), _terminator(terminator), _worker_id(worker_id) {}
 
   virtual void do_void() {
-    assert(_promotion_manager != NULL, "Sanity");
+    assert(_promotion_manager != nullptr, "Sanity");
```

Comment:
I'd prefer this assert was in the constructor.

Reply:
I agree. I further prefer not having an assert and instead using a reference. I also fear doing unrelated changes will be rejected. This is no new code of mine, it is added so that nullptr is used consistently after push-back from Thomas.

Reply:
What would you prefer?
1) keep as is now.
2) move to constructor.
3) use a reference instead.

Reply:
I would probably move it to the constructor, but I don't feel strongly about it.